### PR TITLE
Added conformsToProtocol feature for typhoon matcher

### PR DESCRIPTION
**Pull request by Emailrus**
Related issue https://github.com/appsquickly/Typhoon/issues/454

### CODE REVIEW

File: Source/Configuration/DefinitionOptionConfiguration/Matcher/TyphoonOptionMatcher.m [hunk 1]
```diff
@@ -24,11 +24,13 @@ @interface TyphoonOptionMatch : NSObject
 @property (nonatomic, strong) id value;
 @property (nonatomic) Class memberClass;
 @property (nonatomic) Class kindClass;
+@property (nonatomic) Protocol *proto;
```
Comment (igrekde) [inline Source/Configuration/DefinitionOptionConfiguration/Matcher/TyphoonOptionMatcher.m]: Let's use the full name here - `protocol` (or, if it's reserved, something like `conformProtocol`.

File: Source/Configuration/DefinitionOptionConfiguration/Matcher/TyphoonOptionMatcher.m [hunk 2]
```diff
@@ -141,7 +156,8 @@ - (TyphoonOptionMatch *)matchForValue:(id)value
         BOOL isEqual = (match.value && [match.value isEqual:value]) || ([match.value isEqual:[TyphoonOptionMatchNilValue class]] && !value);
         BOOL isKind = (match.kindClass && [value isKindOfClass:match.kindClass]);
         BOOL isMember = (match.memberClass && [value isMemberOfClass:match.memberClass]);
-        if (isEqual || isKind || isMember) {
+        BOOL isConforms = (match.proto && [value conformsToProtocol:match.proto]);
```
Comment (igrekde) [inline Source/Configuration/DefinitionOptionConfiguration/Matcher/TyphoonOptionMatcher.m]: I think, that `doesConform` is a more suitable name.

**Comment by igrekde** [comment]
@Emailrus Thanks for contributing! I've just reviewed your PR - it looks awesome! 🎉

Just a few style objections from me - and we can merge it.

**Comment by alexgarbarev** [comment]
Thanks @Emailrus for contribution and @igrekde for review. I've reviews and totally agreed with @igrekde :-) Let's merge right after minor spelling fixes

**Comment by Emailrus** [comment]
Done and thanks for the lib

**Comment by igrekde** [comment]
Thanks!

**Pull request merged**